Wed May 11 22:58:46 CEST 2005  Norbert@Nemec-online.de
  * marvin

Mon Apr 11 18:25:51 CEST 2005  Norbert@Nemec-online.de
  * utils changes

Mon Apr 11 18:18:00 CEST 2005  Norbert.Nemec@physik.uni-regensburg.de
  * misc

Index: siesta/Pseudo/atom/dsolv1.f
===================================================================
--- siesta.orig/Pseudo/atom/dsolv1.f	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/dsolv1.f	2006-04-12 17:25:26.000000000 +0200
@@ -103,11 +103,11 @@
             call tridib(nrm,eps,d,sd,sd2,bl,bu,1,nmax(i,j),e,ind,ierr,
      &                  rv4,rv5)
             if (ierr .ne. 0) write(6,9000) ierr
- 9000       format(/' error in tridib ****** ierr =',i3,/)
+ 9000       format(/' error in tridib ******* ierr =',i5,/)
             call tinvit(nrm,nrm,d,sd,sd2,nmax(i,j),e,ind,z,ierr,rv1,rv2,
      &                  rv3,rv4,rv5)
             if (ierr .ne. 0) write(6,9010) ierr
- 9010       format(/' error in tinvit ****** ierr =',i3,/)
+ 9010       format(/' error in tinvit ******* ierr =',i5,/)
 c
 c   Save the energy levels and add to charge density.
 c
Index: siesta/Pseudo/atom/makefile
===================================================================
--- siesta.orig/Pseudo/atom/makefile	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/makefile	2006-04-12 17:25:26.000000000 +0200
@@ -7,7 +7,7 @@
 #----------------------------------------------$$$$$
 # EDIT the following lines to suit your system
 #
-FC= f77
+FC= ifort
 FFLAGS= -O
 LDFLAGS=
 AUX_OBJS= auxf95.o   # Change if you do not have
Index: siesta/Pseudo/atom/Tutorial/ae.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/ae.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/ae.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_s=`test -f AEWFNR0 && echo 1 || echo 0`
 there_is_p=`test -f AEWFNR1 && echo 1 || echo 0`
 there_is_d=`test -f AEWFNR2 && echo 1 || echo 0`
Index: siesta/Pseudo/atom/Tutorial/ae.sh
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/ae.sh	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/ae.sh	2006-04-12 17:25:26.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh -f
+#!/bin/sh
 #
 # ae.sh -- Script to run all-electron atomic calculations
 #
Index: siesta/Pseudo/atom/Tutorial/charge.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/charge.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/charge.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_ps=`test -f PSCHARGE && echo 1 || echo 0`
 there_is_pt=`test -f PTCHARGE && echo 1 || echo 0`
 #
Index: siesta/Pseudo/atom/Tutorial/coreq.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/coreq.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/coreq.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_core=`test -f COREQ && echo 1 || echo 0`
 #
 if (there_is_core==0) quit
Index: siesta/Pseudo/atom/Tutorial/pg.sh
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/pg.sh	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/pg.sh	2006-04-12 17:25:26.000000000 +0200
@@ -1,13 +1,13 @@
-#!/bin/sh -f
+#!/bin/sh
 #
 # pg.sh -- Script to run pseudopotential generation calculations
 #
 # Usage: pg.sh <name.inp>
 #
-DEFAULT_DIR=../..
+DEFAULT_DIR=$PWD/..
 ATOM_UTILS_DIR=${ATOM_UTILS_DIR:-${DEFAULT_DIR}}
 #
-default="../../../atm"
+default="atm"
 prog=${ATOM_PROGRAM:-$default}
 #
 if [ "$#" != 1 ] 
@@ -22,8 +22,9 @@
 #
 if [ -d $name ] 
 then
-	echo "Directory $name exists. Please delete it first"
-	exit
+	rm -rf $name
+#	echo "Directory $name exists. Please delete it first"
+#	exit
 fi
 #
 mkdir $name ; cd $name
Index: siesta/Pseudo/atom/Tutorial/pots.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/pots.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/pots.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_s=`test -f PSPOTR0 && echo 1 || echo 0`
 there_is_p=`test -f PSPOTR1 && echo 1 || echo 0`
 there_is_d=`test -f PSPOTR2 && echo 1 || echo 0`
Index: siesta/Pseudo/atom/Tutorial/pseudo.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/pseudo.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/pseudo.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 #
 # Driver for PS analysis
 # Note the very kludgeish mechanism to find out what is available
Index: siesta/Pseudo/atom/Tutorial/pt.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/pt.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/pt.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_s=`test -f AEWFNR0 && echo 1 || echo 0`
 there_is_p=`test -f AEWFNR1 && echo 1 || echo 0`
 there_is_d=`test -f AEWFNR2 && echo 1 || echo 0`
Index: siesta/Pseudo/atom/Tutorial/pt.sh
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/pt.sh	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/pt.sh	2006-04-12 17:25:26.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh -f
+#!/bin/sh
 #
 # pt.sh -- Script to run pseudopotential test calculations
 #
@@ -6,10 +6,10 @@
 #
 # Make sure that atm is in path
 #
-DEFAULT_DIR=../..
+DEFAULT_DIR=$PWD/..
 ATOM_UTILS_DIR=${ATOM_UTILS_DIR:-${DEFAULT_DIR}}
 #
-default="../../../atm"
+default="atm"
 prog=${ATOM_PROGRAM:-$default}
 #
 if [ "$#" != 2 ] 
@@ -27,8 +27,9 @@
 #
 if [ -d $name ] 
 then
-	echo "Directory $name exists. Please delete it first"
-	exit
+	rm -r $name
+	#echo "Directory $name exists. Please delete it first"
+	#exit
 fi
 #
 mkdir $name ; cd $name
@@ -45,5 +46,3 @@
             cp -f ${ATOM_UTILS_DIR}/$i.gps .
             cp -f ${ATOM_UTILS_DIR}/$i.gplot .
 done
-
-
Index: siesta/Pseudo/atom/Tutorial/scrpots.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/scrpots.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/scrpots.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 #
 # Plots bare and screened pseudopotentials
 #
Index: siesta/Pseudo/atom/Tutorial/vcharge.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/vcharge.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/vcharge.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_ps=`test -f PSCHARGE && echo 1 || echo 0`
 there_is_pt=`test -f PTCHARGE && echo 1 || echo 0`
 #
Index: siesta/Pseudo/atom/Tutorial/vspin.gplot
===================================================================
--- siesta.orig/Pseudo/atom/Tutorial/vspin.gplot	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Pseudo/atom/Tutorial/vspin.gplot	2006-04-12 17:25:26.000000000 +0200
@@ -1,3 +1,5 @@
+#!/usr/bin/gnuplot -persist
+
 there_is_ps=`test -f PSCHARGE && echo 1 || echo 0`
 there_is_pt=`test -f PTCHARGE && echo 1 || echo 0`
 #
Index: siesta/Src/arch.make
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ siesta/Src/arch.make	2006-04-12 18:46:18.000000000 +0200
@@ -0,0 +1,41 @@
+SIESTA_ARCH=intel-mkl
+#
+# Intel fortran compiler for linux with mkl optimized blas and lapack
+#
+# Be sure to experiment with different optimization options.
+# You have quite a number of combinations to try...
+#
+FC=ifort
+#
+FFLAGS= -w -mp -tpp7 -O3
+FFLAGS_DEBUG= -g
+LDFLAGS=-Vaxlib
+COMP_LIBS=
+RANLIB=echo
+#
+NETCDF_LIBS=
+NETCDF_INTERFACE=
+DEFS_CDF=
+#
+MPI_INTERFACE=
+MPI_INCLUDE=
+DEFS_MPI=
+#
+MKL=/usr/local/intel_mkl_61
+GUIDE=$(MKL)/lib/32/libguide.a
+LAPACK=$(MKL)/lib/32/libmkl_lapack.a
+BLAS=$(MKL)/lib/32/libmkl_ia32.a
+#G2C=/usr/lib/gcc-lib/i386-redhat-linux/2.96/libg2c.a
+LIBS=$(LAPACK) $(BLAS) $(G2C) $(GUIDE)  -lpthread
+SYS=bsd
+DEFS= $(DEFS_CDF) $(DEFS_MPI)
+#
+.F.o:
+	$(FC) -c $(FFLAGS)  $(DEFS) $<
+.f.o:
+	$(FC) -c $(FFLAGS)  $<
+.F90.o:
+	$(FC) -c $(FFLAGS)  $(DEFS) $<
+.f90.o:
+	$(FC) -c $(FFLAGS)  $<
+#
Index: siesta/Src/Makefile
===================================================================
--- siesta.orig/Src/Makefile	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Src/Makefile	2006-04-12 18:46:18.000000000 +0200
@@ -44,8 +44,8 @@
 	@echo "If this is not what you want, create the right"
 	@echo "arch.make file using the models in Sys"
 	@echo
-	@echo "Hit ^C to abort..."
-	@sleep 2
+#	@echo "Hit ^C to abort..."
+#	@sleep 2
 #
 SYSOBJ=$(SYS).o pxf.o
 #
Index: siesta/Util/eig2dos.f
===================================================================
--- siesta.orig/Util/eig2dos.f	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Util/eig2dos.f	2006-04-12 17:25:26.000000000 +0200
@@ -126,11 +126,11 @@
       write(6,"(a,2f10.4,a)") '# Number of electrons = ', sta, sum,
      .                        ' per cell'
       write(6,"(a)") '#'
-      write(6,"(a)") '#        E            N'
+      write(6,"(a)") '#        N            E'
       write(6,"(2a)") '# ------------------------------------------',
      .                '--------------------------------'
 
-      write(6,"(2f14.6)") (emin + (ie-1)*eincr, dos(ie), ie = 1, ne)
+      write(6,"(2f14.6)") (dos(ie), emin + (ie-1)*eincr, ie = 1, ne)
 
       stop
       end
Index: siesta/Util/gnubands.f
===================================================================
--- siesta.orig/Util/gnubands.f	2006-02-24 17:21:34.000000000 +0100
+++ siesta/Util/gnubands.f	2006-04-12 17:25:26.000000000 +0200
@@ -23,7 +23,7 @@
       implicit none
 
       integer           maxk, maxb, maxs
-      parameter         (maxk=1000, maxb=100, maxs=2)
+      parameter         (maxk=10000, maxb=1000, maxs=2)
 
       integer           nk, nspin, nband, ik, is, ib
       double precision  e(maxb, maxs, maxk), k(maxk)
@@ -48,18 +48,18 @@
      .                '--------------------------------'
       write(6,"(a,f10.4)")  '# E_F               = ', ef
       write(6,"(a,2f10.4)") '# k_min, k_max      = ', kmin, kmax
-      write(6,"(a,2f10.4)") '# E_min, E_max      = ', emin, emax
+      write(6,"(a,2f10.4)") '# E_min, E_max      = ', emin-ef, emax-ef
       write(6,"(a,3i6)")    '# Nbands, Nspin, Nk = ', nband, nspin, nk
       write(6,"(a)") '#'
       write(6,"(a)") '#        k            E'
       write(6,"(2a)") '# ------------------------------------------',
      .                '--------------------------------'
 
-      read(5,*) (k(ik),((e(ib,is,ik),ib=1,nband), is=1,nspin), ik=1,nk)
+      read(5,*) (k(ik), ((e(ib,is,ik),ib=1,nband), is=1,nspin), ik=1,nk)
      
       do is = 1, nspin
         do ib = 1, nband
-           write(6,"(2f14.6)") ( k(ik), e(ib,is,ik), ik = 1, nk)
+           write(6,"(2f14.6)") ( k(ik), e(ib,is,ik)-ef, ik = 1, nk)
            write(6,"(/)")
         enddo
       enddo
